src/clarity-angular/datagrid/datagrid-column.spec.ts

```diff
@@ -196,32 +229,46 @@ export default function(): void {
+        describe("View filters", function () {
+            it("doesn't display any filter by default", function () {
+                this.context = this.create(DatagridColumn, SimpleTest, [Sort, FiltersProvider]);
                 expect(this.context.clarityElement.querySelector("clr-dg-filter")).toBeNull();
             });
 
-            it("displays a string filter when using a property name", function() {
-                this.context = this.create(DatagridColumn, SimpleTest, [Sort, Filters]);
+            it("displays a string filter when using a property name", function () {
+                this.context = this.create(DatagridColumn, SimpleTest, [Sort, FiltersProvider]);
                 this.context.testComponent.field = "test";
                 this.context.detectChanges();
                 expect(this.context.clarityElement.querySelector("clr-dg-string-filter")).not.toBeNull();
             });
 
-            it("projects custom filters outside of the title", function() {
-                this.context = this.create(DatagridColumn, FilterTest, [Sort, Filters]);
+            it("projects custom filters outside of the title", function () {
+                this.context = this.create(DatagridColumn, FilterTest, [Sort, FiltersProvider]);
                 expect(this.context.clarityElement.querySelector(".my-filter")).not.toBeNull();
                 let title = this.context.clarityElement.querySelector(".datagrid-column-title");
                 expect(title.querySelector(".my-filter")).toBeNull();
             });
 
-            it("projects custom string filters outside of the title", function() {
-                this.context = this.create(DatagridColumn, StringFilterTest, [Sort, Filters]);
+            it("projects custom string filters outside of the title", function () {
+                this.context = this.create(DatagridColumn, StringFilterTest, [Sort, FiltersProvider]);
                 expect(this.context.clarityElement.querySelector(".my-string-filter")).not.toBeNull();
                 let title = this.context.clarityElement.querySelector(".datagrid-column-title");
                 expect(title.querySelector(".my-string-filter")).toBeNull();
             });
+
+            it("un-registers the correct filter", function () {
+                this.context = this.create(DatagridColumn, UnregisterTest, [Sort, FiltersProvider]);
+                this.context.testComponent.show = true;
+                this.context.clarityDirective.filters.add(new TestFilter());
+                this.context.clarityDirective.filters.add(new TestFilter());
+                this.context.detectChanges();
+                this.context.testComponent.show = false;
+                this.context.detectChanges();
+                let filters = this.context.clarityDirective.filters.getActiveFilters();
+                filters.map((filter: TestFilter) => {
+                    expect(filter instanceof TestFilter).toEqual(true);
+                });
```

Comment:
You're not checking that they're still here. For all you know, `filters` might be an empty array and the test would still pass. 😛